Adds continuous integration setup with tox and coverage

lukpueh opened the pull request:
- Decorates not implemented unittests to be skipped
- Updates tox.ini to install requirements, run coverage and call `runtests.py` instead of `setup.py test`.
- Updates .travis.yml to use tox and coveralls. *Running coveralls independently of Travis CI would require to "provide a repo_token in .coveralls.yml"*

lukpueh committed:
Adds skiptest decorator to unimplemented unittests

lukpueh committed:
Updates tox.ini to not use setup.py

- Integrating tox with setup.py test is as of Oct 2016 discouraged
  Now directly calls test/runtests.py
- Adds optional requirements to tox.ini
- Comments out coveralls (just for now)

lukpueh committed:
Updates .travis.yml to use tox

lukpueh committed:
Modifies tox.ini to run testrunner with coverage

After running tox use "coverage report -m" to see a detailed
coverage report.

lukpueh committed:
Modifies .travis.yml to run coveralls on pass

Coveralls is triggered explicitely on Travis CI and not integrated
with tox, to only run coveralls when running tox on Travis CI
and not locally.

lukpueh commented:
Let's make this a 100%!

SantiagoTorres commented on .travis.yml:
```diff
@@ -1,10 +1,18 @@
     tox
+
+after_success:
+    coveralls
```
it'd be nice if we had a newline here

SantiagoTorres commented on test/test_toto_verify.py:
```diff
@@ -8,6 +8,8 @@
 class Test(unittest.TestCase):
 
+  @unittest.skip("Missing test implementation")
+
```
can we remove this whitespace?

SantiagoTorres commented on tox.ini:
```diff
@@ -5,15 +5,22 @@
+    cryptography
+    pytest
+    PyNaCl
+    coverage
```
maybe we could create a file called requirements-ci.txt and add these there. Just keep -rrequirements-ci.txt...

SantiagoTorres commented on tox.ini:
```diff
@@ -5,15 +5,22 @@
-    coveralls
+setenv =
+    SIMPLE_SETTINGS=toto.settings
+    # PYTHONPATH = {toxinidir}
```
Do we need this commented out?

lukpueh committed:
Adds newline at end of .travis.yml

lukpueh committed:
Removes newline in test_toto_verify

lukpueh committed:
Moves requirements to requirements-ci.txt

- Hardcoded test requirements from tox.ini are now in
  requirements-ci.txt
- Removes unneeded comment from tox.ini

SantiagoTorres commented:
@lukpueh I think this is ready to merge? 👍

lukpueh commented:
Thanks for the review, @SantiagoTorres!
